src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/Converters/VectorConverter.cs

```diff
@@ -0,0 +1,222 @@
+                else if (typeof(T) == typeof(ulong))
+                    writer.WriteValue((ulong)(object)value); 
+            
+                else if (typeof(T) == typeof(sbyte))
+                    writer.WriteValue((sbyte)(object)value);  
+                else if (typeof(T) == typeof(short))
+                    writer.WriteValue((short)(object)value); 
+                else if (typeof(T) == typeof(int))
+                    writer.WriteValue((int)(object)value);  
+                else if (typeof(T) == typeof(long))
+                    writer.WriteValue((long)(object)value);
+                else
+                    writer.WriteValue(value);
+            }
+        }
+    }
+    
+    public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+    {
+        if (value == null)
+        {
+            writer.WriteNull();
+            return;
+        }
+        
+        writer.WriteStartObject();
+        writer.WritePropertyName(Sparrow.Global.Constants.Naming.VectorPropertyName);
+        
+        bool canWriteInNativeForm = CanWriteVectorInNativeForm(writer);
+        
+        if (canWriteInNativeForm == false)
+            writer.WriteStartArray();
+
+        // For known types we can cast vector, otherwise we will use reflection
+        switch (value)
+        {
+            case RavenVector<float> v:
+                WriteVectorArray(writer, v, canWriteInNativeForm);
+                break;
+            case RavenVector<double> v:
+                WriteVectorArray(writer, v, canWriteInNativeForm);
+                break;
+            case RavenVector<decimal> v:
+                WriteVectorArray(writer, v, canWriteInNativeForm);
+                break;
+            
+            case RavenVector<byte> v:
+                WriteVectorArray(writer, v, canWriteInNativeForm);
+                break;
+            case RavenVector<ushort> v:
+                WriteVectorArray(writer, v, canWriteInNativeForm);
+                break;
+            case RavenVector<uint> v:
+                WriteVectorArray(writer, v, canWriteInNativeForm);
+                break;
+            case RavenVector<ulong> v:
+                WriteVectorArray(writer, v, canWriteInNativeForm);
+                break;
+            case RavenVector<sbyte> v:
+                WriteVectorArray(writer, v, canWriteInNativeForm);
+                break;
+            case RavenVector<short> v:
+                WriteVectorArray(writer, v, canWriteInNativeForm);
+                break;
+            case RavenVector<int> v:
+                WriteVectorArray(writer, v, canWriteInNativeForm);
+                break;
+            case RavenVector<long> v:
+                WriteVectorArray(writer, v, canWriteInNativeForm);
+                break;
+#if NET6_0_OR_GREATER
+            case RavenVector<Half> v:
+                WriteVectorArray(writer, v, canWriteInNativeForm);
+                break;
+#endif
+            default:
```

Comment:
So it `default` we assume it must be `IEnumerable`, right? Why not handling it explcitly via `case IEnumerable:` ?

Reply:
Added a `IEnumerable` case and added an exception when we don't match a known type.